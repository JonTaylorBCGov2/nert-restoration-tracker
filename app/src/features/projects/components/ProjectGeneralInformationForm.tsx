import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import CustomTextField from 'components/fields/CustomTextField';
import StartEndDateFields from 'components/fields/StartEndDateFields';
import { useFormikContext } from 'formik';
import React from 'react';
import yup from 'utils/YupSchema';

export interface IProjectGeneralInformationForm {
  project: {
    project_name: string;
    start_date: string;
    end_date: string;
    objectives: string;
  };
}

export const ProjectGeneralInformationFormInitialValues: IProjectGeneralInformationForm = {
  project: {
    project_name: '',
    start_date: '',
    end_date: '',
    objectives: ''
  }
};

export const ProjectGeneralInformationFormYupSchema = yup.object().shape({
  project: yup.object().shape({
    project_name: yup.string().max(300, 'Cannot exceed 300 characters').required('Required'),
    start_date: yup.string().isValidDateString().required('Required'),
    end_date: yup.string().isValidDateString().isEndDateAfterStartDate('start_date'),
    objectives: yup
      .string()
      .max(3000, 'Cannot exceed 3000 characters')
      .required('You must provide objectives for the project')
  })
});
/**
 * Create project - General information section
 *
 * @return {*}
 */
const ProjectGeneralInformationForm: React.FC = () => {
  const formikProps = useFormikContext<IProjectGeneralInformationForm>();

  return (
    <Grid container spacing={3}>
      <Grid item xs={12} md={9}>
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <TextField fullWidth required variant="outlined" label="Project Name" id="project.project_name"></TextField>
          </Grid>
          <StartEndDateFields
            formikProps={formikProps}
            startName={'project.start_date'}
            endName={'project.end_date'}
            startRequired={true}
            endRequired={false}
          />
          <Grid item xs={12}>
            <Grid item xs={12}>
              <CustomTextField
                name="project.objectives"
                label="Objectives"
                other={{ multiline: true, required: true, rows: 4 }}
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

export default ProjectGeneralInformationForm;
