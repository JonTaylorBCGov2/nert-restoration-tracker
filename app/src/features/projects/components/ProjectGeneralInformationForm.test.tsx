import { render } from '@testing-library/react';
import { Formik } from 'formik';
import React from 'react';
import ProjectGeneralInformationForm, {
  IProjectGeneralInformationForm,
  ProjectGeneralInformationFormInitialValues,
  ProjectGeneralInformationFormYupSchema
} from './ProjectGeneralInformationForm';

describe('ProjectGeneralInformationForm', () => {
  it('renders correctly with default empty values', () => {
    const { asFragment } = render(
      <Formik
        initialValues={ProjectGeneralInformationFormInitialValues}
        validationSchema={ProjectGeneralInformationFormYupSchema}
        validateOnBlur={true}
        validateOnChange={false}
        onSubmit={async () => {}}>
        {() => <ProjectGeneralInformationForm />}
      </Formik>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with existing details values', () => {
    const existingFormValues: IProjectGeneralInformationForm = {
      project_name: 'name 1',
      start_date: '2021-03-14',
      end_date: '2021-04-14',
      objectives: 'my objectives'
    };

    const { asFragment } = render(
      <Formik
        initialValues={existingFormValues}
        validationSchema={ProjectGeneralInformationFormYupSchema}
        validateOnBlur={true}
        validateOnChange={false}
        onSubmit={async () => {}}>
        {() => <ProjectGeneralInformationForm />}
      </Formik>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
