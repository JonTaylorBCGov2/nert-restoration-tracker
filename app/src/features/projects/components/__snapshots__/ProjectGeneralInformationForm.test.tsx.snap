// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectGeneralInformationForm renders correctly with default empty values 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink="false"
                  for="project_name"
                  id="project_name-label"
                >
                  Project Name
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="project_name"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-7"
                    >
                      <span>
                        Project Name *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-required Mui-required"
                    data-shrink="true"
                    for="start_date"
                    id="start_date-label"
                  >
                    Start Date
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="start-date"
                      id="start_date"
                      max="2100-12-31"
                      min="1900-01-01"
                      name="start_date"
                      required=""
                      type="date"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
                      >
                        <span>
                          Start Date *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                    data-shrink="true"
                    for="end_date"
                    id="end_date-label"
                  >
                    End Date
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="end-date"
                      id="end_date"
                      max="2100-12-31"
                      min="1900-01-01"
                      name="end_date"
                      type="date"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
                      >
                        <span>
                          End Date
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink="false"
                  for="objectives"
                  id="objectives-label"
                >
                  Objectives
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    id="objectives"
                    name="objectives"
                    required=""
                    rows="4"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-7"
                    >
                      <span>
                        Objectives *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <fieldset
            class="MuiBox-root MuiBox-root-9"
          >
            <legend
              class="MuiTypography-root MuiTypography-body1"
            >
              IUCN Conservation Actions Classifications (Optional)
            </legend>
            <div
              class="MuiBox-root MuiBox-root-10"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                Conservation actions are specific actions or sets of tasks undertaken by project staff designed to reach each of the project's objectives.
              </p>
            </div>
            <ul
              class="MuiList-root MuiList-padding"
            >
              <li
                class="MuiListItem-container"
              >
                <div
                  class="MuiListItem-root makeStyles-customListItem-3 MuiListItem-gutters MuiListItem-secondaryAction"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="iucn-classification"
                        >
                          Classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="iucn-classification iucn-classification-select"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="iucn-classification-select"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-7"
                            >
                              <span>
                                Classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="sub-classification-1"
                        >
                          Sub-classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="sub-classification-1 icun-subclass-select-1"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="icun-subclass-select-1"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-7"
                            >
                              <span>
                                Sub-classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="sub-classification-2"
                        >
                          Sub-classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="sub-classification-2 iucn-subclass-select-2"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="iucn-subclass-select-2"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-7"
                            >
                              <span>
                                Sub-classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiListItemSecondaryAction-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        role="presentation"
                        style="width: 1.5rem; height: 1.5rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </li>
            </ul>
            <div
              class="MuiBox-root MuiBox-root-11"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      role="presentation"
                      style="width: 1.5rem; height: 1.5rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                  Add Classification
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ProjectGeneralInformationForm renders correctly with existing details values 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink="false"
                  for="project_name"
                  id="project_name-label"
                >
                  Project Name
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="project_name"
                    required=""
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-18"
                    >
                      <span>
                        Project Name *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                    data-shrink="true"
                    for="start_date"
                    id="start_date-label"
                  >
                    Start Date
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="start-date"
                      id="start_date"
                      max="2100-12-31"
                      min="1900-01-01"
                      name="start_date"
                      required=""
                      type="date"
                      value="2021-03-14"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
                      >
                        <span>
                          Start Date *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                    for="end_date"
                    id="end_date-label"
                  >
                    End Date
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      data-testid="end-date"
                      id="end_date"
                      max="2100-12-31"
                      min="1900-01-01"
                      name="end_date"
                      type="date"
                      value="2021-04-14"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
                      >
                        <span>
                          End Date
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink="false"
                  for="objectives"
                  id="objectives-label"
                >
                  Objectives
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    id="objectives"
                    name="objectives"
                    required=""
                    rows="4"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-18"
                    >
                      <span>
                        Objectives *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <fieldset
            class="MuiBox-root MuiBox-root-20"
          >
            <legend
              class="MuiTypography-root MuiTypography-body1"
            >
              IUCN Conservation Actions Classifications (Optional)
            </legend>
            <div
              class="MuiBox-root MuiBox-root-21"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                Conservation actions are specific actions or sets of tasks undertaken by project staff designed to reach each of the project's objectives.
              </p>
            </div>
            <ul
              class="MuiList-root MuiList-padding"
            >
              <li
                class="MuiListItem-container"
              >
                <div
                  class="MuiListItem-root makeStyles-customListItem-14 MuiListItem-gutters MuiListItem-secondaryAction"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="iucn-classification"
                        >
                          Classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="iucn-classification iucn-classification-select"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="iucn-classification-select"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-18"
                            >
                              <span>
                                Classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="sub-classification-1"
                        >
                          Sub-classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="sub-classification-1 icun-subclass-select-1"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="icun-subclass-select-1"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-18"
                            >
                              <span>
                                Sub-classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                          data-shrink="false"
                          id="sub-classification-2"
                        >
                          Sub-classification
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                        >
                          <div
                            aria-haspopup="listbox"
                            aria-labelledby="sub-classification-2 iucn-subclass-select-2"
                            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                            id="iucn-subclass-select-2"
                            role="button"
                            tabindex="0"
                          >
                            <span>
                              ​
                            </span>
                          </div>
                          <input
                            aria-hidden="true"
                            class="MuiSelect-nativeInput"
                            tabindex="-1"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="PrivateNotchedOutline-legendLabelled-18"
                            >
                              <span>
                                Sub-classification
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiListItemSecondaryAction-root"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        role="presentation"
                        style="width: 1.5rem; height: 1.5rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </li>
            </ul>
            <div
              class="MuiBox-root MuiBox-root-22"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      role="presentation"
                      style="width: 1.5rem; height: 1.5rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </span>
                  Add Classification
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
